dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:stack_trace/stack_trace.dart';
+
+import 'protocol/sentry_stack_frame.dart';
+
+class SentryStackTraceFactory {
+  const SentryStackTraceFactory();
+
+  List<SentryStackFrame> getStackFrames(dynamic stackTrace) {
+    if (stackTrace == null) return [];
+
+    final chain = stackTrace is StackTrace
+        ? Chain.forTrace(stackTrace)
+        : Chain.parse(stackTrace as String);
+
+    final frames = <SentryStackFrame>[];
+    for (var t = 0; t < chain.traces.length; t += 1) {
+      final encodedFrames =
+          chain.traces[t].frames.map((f) => encodeStackTraceFrame(f));
+
+      frames.addAll(encodedFrames);
+
+      if (t < chain.traces.length - 1) {
+        frames.add(SentryStackFrame.asynchronousGapFrameJson);
+      }
+    }
+
+    return frames.reversed.toList();
+  }
+
+  SentryStackFrame encodeStackTraceFrame(Frame frame) {
```

Comment:
should this be private too

Reply:
Is `@visibleForTest` acceptable?

Reply:
yeah we used in Java as well, but if possible to test `getStackFrames` that forwards the call to `encodeStackTraceFrame`, it'd not be needed.
If the test already exists and it works, let's visibleForTest

Reply:
✅